### ISSUE TITLE
[Rule RSRCH]: element_id_unique should be a "Needs review" rather than "Violation"

### ISSUE THREAD

**Issue by jclintonibm** [IBMa/equal-access#2325]
### Rule description

The rule `element_id_unique` should be a "Needs review" rather than "Violation" since non-unique IDs are not strictly an accessibility issue, they are a more general HTML parsing issue.

An example is where an external module is supplying SVGs along with functions to compose them into re-usable fragments. An SVG fragment of one is:
```
<svg xmlns="http://www.w3.org/2000/svg" x...>
  <defs>
    <linearGradient id="jira-linear-gradient" ...>
      ...
    </linearGradient>
  </defs>
  <path ... fill="url(#jira-linear-gradient)" />
</svg>

```

These can be composed into SVG groups by the external module e.g.:

<img width="200" height="58" alt="Image" src="https://github.com/user-attachments/assets/d6da1845-ad0d-48ee-a5a8-6591cada1e6c" />

The IBM Accessibility tool triggers a violation when there are multiple copies of this same SVG and therefore there are multiple occurrences of the linearGradient ID "jira-linear-gradient" (or multiples of any of the hundreds of SVGs provided by the external module that include a `defs` object).

In addition, a page in our application may contain dozens or hundreds of these "violations" if an SVG occurs more than once e.g.:

<img width="140" height="266" alt="Image" src="https://github.com/user-attachments/assets/5cfd56a5-7291-41c9-a172-3b418fc4821f" />

This means it's easy to miss genuine accessibility issues in the noise.

When triggered the rule itself gives the following information:
> Note: The [4.1.1 Parsing](https://www.w3.org/WAI/WCAG21/Understanding/parsing.html) requirement was removed from WCAG. However, the Checker will report an issue against another accessibility requirement when it detects it causes an accessibility issue.

However, no other accessibility requirements are violated by these SVGs.

### Relevant requirement(s)

4.1.1 Parsing

**Comment by asfordmatt** [comment]
I agree, this is a spec issue but not an accessibility violation. It has caused us a lot of noise.

**Comment by tombrunet** [comment]
Changing these to "Needs review" isn't the right answer because this is a violation of HTML spec - it's bad code. As noted in Slack, these id references aren't contained to the SVG, so the references in one SVG are actually pointing to the other SVG. It just so happens that the referenced instructions are identical so that you can't tell the difference, but they aren't referencing the intended instruction. See https://btrem.com/2021/02-svg-id.

As I noted, you can contain this correctly using template to put it in a shadow dom so that the id references are contained within that shadow dom.

We'll look into scoping this tighter to only highlight it when it's strictly an accessibility error, but that doesn't mean that the id references are correct.

**Comment by jclintonibm** [comment]
> As I noted, you can contain this correctly using template to put it in a shadow dom

As I noted in the thread you refer to, the SVGs and their container are provided by an external module and we can have dozens to hundreds of these in a page. Doing what you suggest would involve us rendering the page then traversing the DOM to locate the SVGs and manipulating the DOM to wrap each SVG in a shadow DOM. Given that this will make no difference to the user visually or to the accessibility of the page, we would find it hard to justify this post processing just to work around the accessibility tool.

> We'll look into scoping this tighter to only highlight it when it's strictly an accessibility error, but that doesn't mean that the id references are correct.

Agreed which is why we have visual diff tests to ensure the SVGs are rendered correctly in our pages. However, as the tool itself notes, HTML parsing issues are not accessibility issues and since this is an accessibility tool, it should only be raising violations when there is an actual accessibility violation.

**Comment by dehru** [comment]
I agree with @jclintonibm - html spec violations and accessibility issues are different.  There are likely plenty of accessible applications that have spec violations.

**Comment by shunguoy** [comment]
triage: change to "NR" now. In the future, we need to check 'labelledby', 'describedby', etc., to ensure the ID pointed to exists and is unique.
